src/title-dropdown/title-dropdown.component.html

```diff
@@ -1,7 +1,9 @@
-<div dropdown>
-  <button class="btn title-dropdown-toggle" dropdownToggle>
+<div dropdown [ngClass]="{disableCursorClass: schema && schema['x_editor_disabled']}">
+  <div [ngClass]="{disableEventsClass: schema && schema['x_editor_disabled']}">
+    <button class="btn title-dropdown-toggle" dropdownToggle>
```

Comment:
instead of `disableEventsClass` you can use `[isDisabled]` of `dropdownToggle` which will do the necessary thing, then you can remove the extra div with `[ngClass]`. Also I think you can even remove the `disableCursorClass`, but I am not 100% sure that `[isDisabled]` will handle this.